src/eins-hwinfo.c

```diff
@@ -0,0 +1,119 @@
+/* Copyright 2018 Endless Mobile, Inc. */
+
+/* This file is part of eos-metrics-instrumentation.
+ *
+ * eos-metrics-instrumentation is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * eos-metrics-instrumentation is distributed in the hope that it will be
+ * useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with eos-metrics-instrumentation.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ */
+#include "eins-hwinfo.h"
+
+#include <eosmetrics/eosmetrics.h>
+#include <gio/gio.h>
+
+/*
+ * Reported at system startup, and every RECORD_DISK_SPACE_INTERVAL seconds
+ * after startup. The payload is two triples of int64, ((ttt)(ttt)).
+ *
+ * The first triple represents the total size, space used, and space available
+ * on the / filesystem, measured in bytes. (On dual-boot installations, this
+ * refers to the Endless OS image file, not the Windows partition it is hosted
+ * on.)
+ *
+ * On split-disk systems, the second triple represents the /var/endless-extra
+ * partition; on single-disk systems, all three values will be 0.
+ *
+ * Space on other user-accessible partitions on the disk, including Windows
+ * partitions on dual-boot systems, is not reported.
+ *
+ * You might think that given any two of these values for a filesystem, you
+ * could derive the third. That's not the case: typically, 5% of space is
+ * reserved (so used + available = 0.95 * total) but this is a tunable
+ * parameter of the filesystem.
+ */
+#define DISK_SPACE_EVENT "b76c645f-d041-41b0-a4d4-c29f86c18638"
+
+#define RECORD_DISK_SPACE_INTERVAL (60u * 60u * 24u)
```

Comment:
Would be nice to have the unit in this define as well, and/or a comment that this amounts to a day.

Reply:
Sure. `G_TIME_SPAN_DAY` exists but is measured in microseconds so I think it's clearer to open-code it here.

Reply:
I have also renamed the existing unitless time constant.